tsconfig.lib.json

```diff
@@ -0,0 +1,9 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "outDir": "./build-lib",
```

Comment:
Why not using the same `build` folder?

Reply:
I can use the same build folder, but it will bloat CDN builds, and CDN builds don't need this stuff.

Reply:
I'd be happy to make the change if you prefer, but I think it will lead to technical debt later.

Reply:
Conceptually, an end user product (build) is a completely separate entity from a library or a header file (build-lib), which is why we distribute .exe files independently from .dll files (or executables independently from shared objects and header files).

Reply:
I don't see a real reason not to make it part of the build. What you can do in order to decrease the size of the file is to remove the `config-sample` file (although I think it won't be included anyway since no one is calling it by default).

Developers who use the library with CDN won't use the `ts` configuration anyway.

Reply:
I will do that then, but I have 2 concerns I wish to address:

- [ ] There is a possibility of filename conflicts from the two build processes. I am willing to ignore this, but I anticipate technical debt.
- [ ] Although the `build` directory is in `.gitignore`, there is an old `build` directory which was added to the repo, probably before the `build` path was added to `.gitignore` . I would like to remove this, since it creates a lot of confusion when developing on the project.

Reply:
I have applied the changes discussed here.

Reply:
> I don't see a real reason not to make it part of the build. What you can do in order to decrease the size of the file is to remove the `config-sample` file (although I think it won't be included anyway since no one is calling it by default).

The config-sample is many orders of magnitude smaller than the output from including a second copy of every source code file as well as type definitions for every source code file. However, it does not consume download time, since nobody will download it. It only consumes disk space, which is a cheap resource.